api/v1beta1/openstackbaremetalset_types.go

```diff
@@ -80,20 +80,62 @@ type OpenStackBaremetalSetProvisioningStatus struct {
+	BaremetalSetCondTypeEmpty ProvisioningState = "Empty"
+	// BaremetalSetCondTypeWaiting - something other than BaremetalHost availability is causing the OpenStackBaremetalSet to wait
+	BaremetalSetCondTypeWaiting ProvisioningState = "Waiting"
+	// BaremetalSetCondTypeProvisioning - one or more BaremetalHosts are provisioning
+	BaremetalSetCondTypeProvisioning ProvisioningState = "Provisioning"
+	// BaremetalSetCondTypeProvisioned - the requested BaremetalHosts count has been satisfied
+	BaremetalSetCondTypeProvisioned ProvisioningState = "Provisioned"
+	// BaremetalSetCondTypeDeprovisioning - one or more BaremetalHosts are deprovisioning
+	BaremetalSetCondTypeDeprovisioning ProvisioningState = "Deprovisioning"
+	// BaremetalSetCondTypeInsufficient - one or more BaremetalHosts not found (either for scale-up or scale-down) to satisfy count request
+	BaremetalSetCondTypeInsufficient ProvisioningState = "Insufficient availability"
+	// BaremetalSetCondTypeError - general catch-all for actual errors
+	BaremetalSetCondTypeError ProvisioningState = "Error"
+
+	//
+	// condition reasones
+	//
+
+	//
+	// metal3
+	//
+
+	// BaremetalHostCondReasonListError - metal3 bmh list objects error
+	BaremetalHostCondReasonListError ConditionReason = "BaremetalHostListError"
+	// BaremetalHostCondReasonGetError - error get metal3 bmh object
+	BaremetalHostCondReasonGetError ConditionReason = "BaremetalHostGetError"
+	// BaremetalHostCondReasonUpdateError - error updating metal3 bmh object
+	BaremetalHostCondReasonUpdateError ConditionReason = "BaremetalHostUpdateError"
+	// BaremetalHostCondReasonCloudInitSecretError - error creating cloud-init secrets for metal3
+	BaremetalHostCondReasonCloudInitSecretError ConditionReason = "BaremetalHostCloudInitSecretError"
+
+	//
+	// osbms
+	//
+
+	// BaremetalSetCondReasonBaremetalHostStatusNotFound - bare metal host status not found
+	BaremetalSetCondReasonBaremetalHostStatusNotFound ConditionReason = "BaremetalHostStatusNotFound"
+	// BaremetalSetCondReasonUserDataSecretDeleteError - error deleting user data secret
+	BaremetalSetCondReasonUserDataSecretDeleteError ConditionReason = "BaremetalSetUserDataSecretDeleteError"
+	// BaremetalSetCondReasonNetworkDataSecretDeleteError - error deleting network data secret
+	BaremetalSetCondReasonNetworkDataSecretDeleteError ConditionReason = "BaremetalSetNetworkDataSecretDeleteError"
+	// BaremetalSetCondReasonScaleDownInsufficientAnnotatedHosts - not enough nodes annotated for deletion
+	BaremetalSetCondReasonScaleDownInsufficientAnnotatedHosts ConditionReason = "BaremetalSetScaleDownInsufficientAnnotatedHosts"
+	// BaremetalSetCondReasonScaleUpInsufficientHosts - not enough nodes for requested host count
+	BaremetalSetCondReasonScaleUpInsufficientHosts ConditionReason = "BaremetalSetScaleUpInsufficientHosts"
+	// BaremetalSetCondReasonProvisioningErrors - errors during bmh provisioning
+	BaremetalSetCondReasonProvisioningErrors ConditionReason = "BaremetalSetProvisioningErrors"
```

Comment:
Should we make the value (but not the variable name) `BaremetalHostProvisioningErrors` to match with the other BMH-related condition reasons that follow?  If this is how you want it though, no worries

Reply:
Nevermind.  I just saw how you use it in the overall status calculation for the BMSet controller.  I think it makes sense how you have it now.